web/elm/src/Views/TopBar.elm

```diff
@@ -121,6 +121,7 @@ pipelineBreadcrumbs session pipeline =
             session.pipelines
                 |> RemoteData.withDefault []
                 |> List.filter (\p -> p.name == pipeline.name && p.teamName == pipeline.teamName)
+                |> List.filter (isPipelineVisible session)
```

Comment:
Purely for my education sake, (and sorry for the stupid question, I forgot everything about elm again). I just had a question about this! I think that `isPipelineVisible` takes in two ..arguments : `isPipelineVisible { favoritedPipelines } p =`? So does `session` satisfy the arguments to pass into `isPipelineVisible`? How is `session` defined, is it a special type (the only definition I can find of Session is in `Application/Models.elm` but it doesnt seem to be the right one 😅 )?

Reply:
`isPipelineVisible`, as you note, is a function that takes in two arguments - one of which is a `record` (~`struct`) with a field `favoritedPipelines`, and the other is a `Pipeline`.

`session` doesn't satisfy both of those arguments - it's only the first one (a `record` with `favoritedPipelines`). One cool thing with Elm is that you can call a function with only a few of the arguments and get back a new function that takes in the remaining ones.

In this case, `isPipelineVisible session` actually gives you back a new function that has one argument, `Pipeline`. Does that make sense?

Reply:
Ohhhh right! I forgot about the thing how you can call a function with a few arguments and get back a function that takes the remaining ones, thanks so much it does make sense now!! 😄